File: api/src/main/java/cn/hirrao/javaee/service/DrugAlertService.java
package cn.hirrao.javaee.service;

import org.springframework.stereotype.Service;

import java.sql.Date;
import java.time.LocalDate;
import java.time.LocalTime;

@Service
public interface DrugAlertService {
    void insertDrugAlert(long alertId, long uid, long drugId, LocalTime alertTime, LocalDate eatTime);

    void updateDrugAlertIsActiveById(long alertId,int isActive);

    void updateDrugAlertEatTimeById(long alertId,Date eatTime);

    void deleteDrugAlertById(long alertId,long uid,long drugId);
}

File: api/src/main/kotlin/cn/hirrao/javaee/service/DrugAlertService.kt
package cn.hirrao.javaee.service

import org.springframework.stereotype.Service
import java.time.LocalDate
import java.time.LocalTime

@Service
interface DrugAlertService {
    fun insertDrugAlert(alertId: Long, uid: Long, drugId: Long, alertTime: LocalTime?, eatTime: LocalDate?)

    fun updateDrugAlertIsActiveById(alertId: Long, isActive: Int)

    fun updateDrugAlertEatTimeById(alertId: Long, eatTime: LocalDate?)

    fun deleteDrugAlertById(alertId: Long, uid: Long, drugId: Long)
}
